File: src/main/java/me/tajam/jext/command/ParameterNumber.java
package me.tajam.jext.command;

import java.util.ArrayList;
import java.util.List;

import org.bukkit.command.CommandSender;

public class ParameterNumber extends Parameter {

  private final List<String> stringValues;
  private String name;

  ParameterNumber(boolean required, Float... suggestedValues) {
    super(required);
    this.stringValues = new ArrayList<>();
    for (Float value : suggestedValues) {
      this.stringValues.add(value.toString());
    }
  }

  ParameterNumber(boolean required, Integer... suggestedValues) {
    super(required);
    this.stringValues = new ArrayList<>();
    for (Integer value : suggestedValues) {
      this.stringValues.add(value.toString());
    }
  }

  ParameterNumber setName(String name) {
    this.name = name;
    return this;
  }

  @Override
  String getName() {
    return this.name == null ? this.name : "number";
  }

  @Override
  List<String> onComplete(String parameter, CommandSender sender) {
    if (parameter.isEmpty()) {
      return stringValues;
    }
    return null;
  }
  
}

File: src/main/java/me/spartacus04/jext/commands/adapter/ParameterNumber.kt
package me.spartacus04.jext.commands.adapter

import org.bukkit.command.CommandSender

/**
 * This class represents a custom number parameter
 *
 * @constructor Creates a new ParameterNumber object
 *
 * @param required A boolean value that represents whether the parameter is required or not
 * @param suggestedValues A list of suggested values
 */
class ParameterNumber internal constructor(required: Boolean, vararg suggestedValues: Number) : Parameter(required) {
    private val stringValues: MutableList<String>
    override var name: String? = null
        get() {
            return if (field == null) field else "number"
        }

    init {
        stringValues = ArrayList()
        for (value in suggestedValues) {
            stringValues.add(value.toString())
        }
    }

    /**
     * This function sets the name of the parameter
     *
     * @param name The name of the parameter
     * @return The current ParameterNumber object
     */
    fun setName(name: String?): ParameterNumber {
        this.name = name
        return this
    }

    override fun onComplete(parameter: String, sender: CommandSender): List<String>? {
        return if (parameter.isEmpty()) {
            stringValues
        } else null
    }
}